src/components/file-input/file-input.tsx

```diff
@@ -0,0 +1,148 @@
+    shadow: true,
+})
+export class FileInput {
+    /**
+     * Specifies the types of files that the dropzone will accept. By default, all file types are accepted.
+     *
+     * For media files, formats can be specified using: `audio/*`, `video/*`, `image/*`.
+     * Unique file type specifiers can also be used, for example: `.jpg`, `.pdf`.
+     * A comma-separated list of file extensions or MIME types is also acceptable, e.g., `image/png, image/jpeg` or
+     * `.png, .jpg, .jpeg`.
+     *
+     * @see [HTML attribute: accept](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/accept) for more
+     * details.
+     */
+    @Prop({ reflect: true })
+    public accept: string = '*';
+
+    /**
+     * Set to `true` to disable file input selection.
+     */
+    @Prop({ reflect: true })
+    public disabled: boolean = false;
+
+    /**
+     * Set to `true` to enable selection of multiple files
+     */
+    @Prop({ reflect: true })
+    public multiple: boolean = false;
+
+    /**
+     * Emitted when files are selected
+     */
+    @Event()
+    filesSelected: EventEmitter<FileInfo[]>;
+
+    @Element()
+    private element: HTMLLimelFileElement;
+
+    private fileInput: HTMLInputElement;
+    private fileInputId = createRandomString();
+
+    public componentDidLoad() {
+        this.fileInput = this.element.shadowRoot.getElementById(
+            this.fileInputId,
+        ) as HTMLInputElement;
+    }
+
+    public render() {
+        return (
+            <Host
+                onClick={this.handleClick}
+                onKeyUp={this.handleKeyUp}
+                onKeyDown={this.handleKeyDown}
+            >
+                <input
+                    hidden={true}
+                    id={this.fileInputId}
+                    onChange={this.handleFileChange}
+                    type="file"
+                    accept={this.accept}
+                    disabled={this.disabled}
+                    multiple={this.multiple}
+                />
+                <slot />
+            </Host>
+        );
+    }
+
+    private handleClick = (event: Event) => {
+        if (this.disabled) {
+            event.stopPropagation();
+            event.preventDefault();
+
+            return;
+        }
+
+        this.triggerFileDialog();
+
+        event.stopPropagation();
+    };
+
+    private handleKeyUp = (event: KeyboardEvent) => {
+        event.stopPropagation();
+        event.preventDefault();
+
+        if (event.code === 'Enter') {
+            this.triggerFileDialog();
+        }
+    };
+
+    private handleKeyDown(event: KeyboardEvent) {
+        if (
+            event.code === 'Tab' ||
+            event.code === 'Backspace' ||
+            event.code === 'Enter'
+        ) {
+            return;
+        }
+
+        event.preventDefault();
+        event.stopPropagation();
+    }
+
+    private triggerFileDialog() {
+        this.fileInput.click();
+    }
+
+    private handleFileChange = (event: Event) => {
```

Comment:
It's possible to remove `event` here, but I'm not sure if you really want to stop the propagation of the change event or not.

Reply:
Hmm, why would we not want to stop the event from propagating? Whom would be interested in that change event you mean?

Reply:
Yeah, partly. I thought the callback could be simplified if we omit the `event` argument, but then we won't be able to stop the event propagation.

I thought about the last question some more just now. But I guess we might as well stop it (because while some other consumer might be interested in these events, it would typically be weird UX if a click of a button had two effects, that is, opening the file selection dialog AND doing the thing it would otherwise do e.g., bring up a snackbar). Let's say we cross that bridge when we get there, if ever :)